UPGRADING.md

```diff
@@ -1,4 +1,113 @@
-# Guide to upgrading from url 0.x to 1.x
+# Upgrade guide
+
+This guide contains steps for upgrading crates in this project between major
+versions. Only the most common issues are covered here. For full details of
+changes, see the [changelog](CHANGELOG.md).
+
+## Upgrading from url 1.x to 2.x
+
+* The minimum supported Rust version is now v1.33.0. Verify that you can bump
+  your library or application to the same MSRV.
+
+* `Url` no longer implements `std::net::ToSocketAddrs`. You will instead need to
+  explicitly convert your `Url` to a type that implements `ToSocketAddrs`.
```

Comment:
Assuming some form of https://github.com/servo/rust-url/pull/535 lands, this should reference it instead.